Compilation error under GCC 10.1.0

Posting this to make sure the patch passes the smell test before I post a PR.

Error
```
     636    make[3]: Entering directory '/tmp/sherbein/spack-stage/spack-stage-flux-core-0.16.0-2pt5mhql2yrkx3di5oibywcxn4paobsb/spack-src/spack-build/src/modules/job-ingest'
     637      CC       job-ingest.lo
     638      CC       validate.lo
     639      CC       worker.lo
     640      CCLD     job-ingest.la
     641    /usr/bin/ld: .libs/validate.o:/tmp/sherbein/spack-stage/spack-stage-flux-core-0.16.0-2pt5mhql2yrkx3di5oibywcxn4paobsb/spack-src/src/modules/job-ingest/validate.h:18: multiple definition of `v'; .libs/job-ingest.o:/tmp/sherbein/spack-stage/spack-stage-flux-core
            -0.16.0-2pt5mhql2yrkx3di5oibywcxn4paobsb/spack-src/src/modules/job-ingest/validate.h:18: first defined here
  >> 642    collect2: error: ld returned 1 exit status
     643    make[3]: *** [Makefile:590: job-ingest.la] Error 1
     644    make[3]: Leaving directory '/tmp/sherbein/spack-stage/spack-stage-flux-core-0.16.0-2pt5mhql2yrkx3di5oibywcxn4paobsb/spack-src/spack-build/src/modules/job-ingest'
     645    make[2]: *** [Makefile:509: all-recursive] Error 1
     646    make[2]: Leaving directory '/tmp/sherbein/spack-stage/spack-stage-flux-core-0.16.0-2pt5mhql2yrkx3di5oibywcxn4paobsb/spack-src/spack-build/src/modules'
     647    make[1]: *** [Makefile:507: all-recursive] Error 1
     648    make[1]: Leaving directory '/tmp/sherbein/spack-stage/spack-stage-flux-core-0.16.0-2pt5mhql2yrkx3di5oibywcxn4paobsb/spack-src/spack-build/src'
```

validate.h:
```C
struct validate *v;

/* Submit jobspec ('buf, 'len') for validation.
 * Future is fulfilled once validation is complete.
 */
flux_future_t *validate_jobspec (struct validate *v, const char *buf, int len);
```

Applying this patch fixes the error:
```diff
diff --git a/src/modules/job-ingest/validate.h b/src/modules/job-ingest/validate.h
index e7fdd078a..2b36f387a 100644
--- a/src/modules/job-ingest/validate.h
+++ b/src/modules/job-ingest/validate.h
@@ -15,7 +15,7 @@

 #include "types.h"

-struct validate *v;
+struct validate;

 /* Submit jobspec ('buf, 'len') for validation.
  * Future is fulfilled once validation is complete.
```

Also running across:
```
flux-terminus.c: In function ‘cmd_kill’:
flux-terminus.c:710:15: error: ‘id’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  710 |     if (!(f = flux_rpc_pack (h, service, rank, 0,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  711 |                              "{s:i s:i s:i}",
      |                              ~~~~~~~~~~~~~~~~
  712 |                              "id", id,
      |                              ~~~~~~~~~
  713 |                              "signal", SIGKILL,
      |                              ~~~~~~~~~~~~~~~~~~
  714 |                              "wait", 1))
      |                              ~~~~~~~~~~
```

Will fix that in an upcoming PR too.

--- a/src/cmd/flux-terminus.c
+++ b/src/cmd/flux-terminus.c
@@ -691,7 +691,7 @@ int cmd_kill (optparse_t *p, int argc, char **argv)
     flux_future_t *f;
     char service [128];
     const char *idstr;
-    int id;
+    int id = -1;
     int optindex = optparse_option_index (p);
     int rank = optparse_get_int (p, "rank", FLUX_NODEID_ANY);
 

--- a/src/modules/job-ingest/validate.h
+++ b/src/modules/job-ingest/validate.h
@@ -15,7 +15,7 @@
 
 #include "types.h"
 
-struct validate *v;
+struct validate;
 
 /* Submit jobspec ('buf, 'len') for validation.
  * Future is fulfilled once validation is complete.

--- a/t/valgrind/valgrind.supp
+++ b/t/valgrind/valgrind.supp
@@ -86,7 +86,7 @@
    match-leak-kinds: definite
    fun:malloc
    fun:dl_open_worker
-   fun:_dl_catch_error
+   fun:_dl_catch_*
    fun:_dl_open
    ...
 }

